docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,237 @@
+package metadata
+
+components: sources: exec: {
+	title: "Exec"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["sidecar"]
+		development:   "beta"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.exec
+			}
+
+			tls: enabled: false
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
+		mode: {
+			description: "The type of exec mechanism."
+			required:    true
+			type: string: {
+				enum: {
+					scheduled: "Scheduled exec mechanism."
+					streaming: "Streaming exec mechanism."
+				}
+				syntax: "literal"
+			}
+		}
+		command: {
+			required:    true
+			description: "The command to be run."
+			type: string: {
+				examples: ["echo", "./myscript.sh"]
+				syntax: "literal"
+			}
+		}
+		arguments: {
+			common:      false
+			description: "Array of any arguments to pass to the command."
+			required:    false
+			type: array: {
+				default: null
+				items: type: [string]: {
+					examples: ["Hello World!", "-la"]
+					syntax: "literal"
+				}
+			}
+		}
```

Comment:
I was thinking to make the `command` parameter a list of strings just like `arguments` is now. Then the command is just `command[0]` and the arguments is `&command[1..]`.

Eventually it might be worth also accepting a single string. The common setup for executing that is to pass it through `$SHELL -c "COMMAND"` which then handles all the command-line parsing and setting up pipes or anything else. However, I don't think that should be the target for this initial feature.

Reply:
I like the idea of command being a list of strings as I am not sure how complex the parsing of arguments might be for different use cases but yes perhaps in the future it could be combined to a single string.

Reply:
I have now removed the arguments config item and changed command to an array.